### ISSUE TITLE
Use scan command in favor of check one for bandit

### ISSUE THREAD

**Issue by jorgepiloto** [ansys/actions#641]
Currently, [bandit] uses the [check] command, see:

https://github.com/ansys/actions/blob/126c7eb0d84802c87341694b7b22a66a02f1430d/check-vulnerabilities/action.yml#L315

However, I started to get this notification:

```console
DEPRECATED: this command (`check`) has been DEPRECATED, and will be unsupported beyond 01 June 2024.

We highly encourage switching to the new `scan` command which is easier to use, more powerful, and can be set up to mimic the deprecated command if required.
```

We should migrate to the new command.

**Comment by RobPasMue** [comment]
Correct - I started working on this but their claim of "being easier to use, more powerful and can be set up to mimic..." is not true The output JSON file had a different structure and required adapting all our scripts. I will continue this work - thanks @jorgepiloto

**Comment by RobPasMue** [comment]
Alsos - the affected package is not ``bandit`` but ``safetyp``

**Comment by RobPasMue** [comment]
Nope - this shouldn't mark the job as failed @PProfizi.

**Comment by RobPasMue** [comment]
We should try to revisit this issue at some point @ansys/pyansys-core - mentioned by @da1910 in #1105

**Comment by moe-ad** [comment]
I looked into this. Besides the need to adapt existing scripts, switching to `safety scan` creates an authentication mess and also requires a major decision regarding cost/pricing to be made. Authentication will be required for both local runs and for CI runs. And if we are sticking with Safety for the long term, this decision **must** be made at some point.

The issue is that `safety scan` requires a Safety account because the results of the scan is always synced with a Safety web GUI that is only reachable via this account (similar to how Codacy works, but not exactly the same), hence the need for the authentication. Besides that, the authentication is also the mechanism by which Safety enforces it's current pricing model:
 - The free tier: single user, single codebase, limited to just 100 scans per month
 - Team: 20 users, 25 codebases, 5000 scans per month, $25 per dev per month or $240 per year (with some extra perks)
 - Enterprise: bespoke pricing ("practically" no limitations wrt users, codebases, and scans, with even more extra perks)

The zero-cost solution would be setting up `ansys/actions/check-vulnerabilities` to accept an API key via an input, and leaving it to the maintainers of each project to open individual Safety accounts to generate API keys. Like I already noted above, this is limited to just 100 scans per month, which might not be sufficient for flagship projects due to the activity in those repos unless we enforce running the action only via a cron job. Also I am not certain PyAnsys projects qualify for this pricing tier:
<img width="389" height="297" alt="Image" src="https://github.com/user-attachments/assets/cd239e90-9730-4f8e-b1b9-112faa438e3c" />

The team subscription is not feasible since there are more than 25 codebases.

With the enterprise tier, SAML-based authentication is supported, which is convenient for local runs. We can also generate a central API key that `ansys/actions/check-vulnerabilities` can use and relieve that responsibility off maintainers.

In conclusion, implementation is not the main blocker anymore. We have bigger fish to fry.

Pinging @MaxJPRey, @RobPasMue, @jorgepiloto, @SMoraisAnsys, @germa89, @inimaz.

**Comment by RobPasMue** [comment]
The switch to the ``scan`` mechanism was postponed in conjunction to your argument, not only because of the script changes. Those are approachable whereas the API key mechanism isn't. I think I presented this internally in the past already.

The current mechanism, despite the warnings, continues to work though.

Switching tools would be the solution in the long run. Either that or accepting the "enterprise level" payment.

However, this was also put on hold until we decide internally (as a company) what tool we might leverage across the board. It isn't a simple decision, nor one we can take on our own I believe

**Comment by moe-ad** [comment]
Could intercepting the `safety check` output and stripping the `DEPRECATED: ...` notes from it be a reasonable partial solution then? To stop tripping people off when they are checking the CI logs at least.

**Comment by RobPasMue** [comment]
Yeah sure, why not =)

**Comment by moe-ad** [comment]
Synced with @MaxJPRey to close this one with #1341. #1342 has been opened for tracking the long term decision.
